layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -353,6 +353,24 @@ initialized with the current filename."
                 ;; ?\a = C-g, ?\e = Esc and C-[
                 ((memq key '(?\a ?\e)) (keyboard-quit))))))))
 
+(defun spacemacs/save-file-as-and-open (filename)
+  "Save current buffer into file FILENAME and open it in a new buffer."
+  (interactive
+   (list (if buffer-file-name
+             (read-file-name "Save file as and open: " buffer-file-name)
+           (read-file-name "Save file as and open: " default-directory))))
+  (or (null filename) (string-equal filename "")
```

Comment:
Can they display the customized message like "Save file as and open: "?

Reply:
I don't think so. But that isn't required because the user will be eventually saving the file with appropriate message from `save-buffer`. If the implementation is simpler and shorter is will be easier to debug and faster to load.